packages/adblocker/test/matching.test.ts

```diff
@@ -448,6 +448,10 @@ describe('#CosmeticFilter.match', () => {
     expect(f`foo.com,sub.test.com##.selector`).not.to.matchHostname('com');
   });
 
+  it('parent domains', () => {
+    expect(f`foo.com>>##+js(foo)`).to.matchHostname('foo.com');
```

Comment:
This test is crucial. So, I'm trying to understand if it matches the de-facto specification (https://github.com/gorhill/uBlock/commit/a483f7955fc8c9c820684c3f55c9f982a7abb2d3):

>   example.com>>##+js(set, iAmEmbeddedInExampleDotCom, true)
>
> The new `>>` specifier means that the scriptlet will not be injected
in `example.com`, but will be injected in all direct and indirect
embedded contexts in pages loaded from `example.com`

If the test is only stating that the filter `foo.com>>##+js(foo)` needs to be considered when visiting a website with hostname `foo.com`, I guess it is correct. Though it would not be correct if it is stating that it would always match (depends if there are subframes).